test/randomized/test_transpiler_equivalence.py

```diff
@@ -262,27 +372,51 @@ def qasm(self):
         """After each circuit operation, it should be possible to build QASM."""
         self.qc.qasm()
 
+    @st.composite
+    def transpiler_conf(draw):
+        opt_level = draw(st.integers(min_value=0, max_value=3))
+        layout_method = draw(st.sampled_from(layout_methods))
+        routing_method = draw(st.sampled_from(routing_methods))
+        scheduling_method = draw(st.sampled_from(scheduling_methods))
+
+        compatible_backends = st.one_of(st.none(), st.sampled_from(mock_backends))
+        if scheduling_method is not None or backend_needs_durations:
+            compatible_backends = st.sampled_from(mock_backends_with_scheduling)
+
+        backend = draw(st.one_of(compatible_backends))
+
+        return (backend, opt_level, layout_method, routing_method, scheduling_method)
+
     @precondition(lambda self: any(isinstance(d[0], Measure) for d in self.qc.data))
-    @rule(
-        backend=st.one_of(st.none(), st.sampled_from(mock_backends)),
-        opt_level=st.integers(min_value=0, max_value=3),
-    )
-    def equivalent_transpile(self, backend, opt_level):
+    @rule(conf=transpiler_conf())
+    def equivalent_transpile(self, conf):
         """Simulate, transpile and simulate the present circuit. Verify that the
         counts are not significantly different before and after transpilation.
 
         """
-
-        print(f"Evaluating circuit at level {opt_level} on {backend}:\n{self.qc.qasm()}")
+        backend, opt_level, layout_method, routing_method, scheduling_method = conf
 
         assume(backend is None or backend.configuration().n_qubits >= len(self.qc.qubits))
 
+        print(
+            f"Evaluating circuit at level {opt_level} on {backend} "
+            f"using layout_method={layout_method} routing_method={routing_method} "
+            f"and scheduling_method={scheduling_method}:\n{self.qc.qasm()}"
+        )
+
         shots = 4096
 
         aer_counts = execute(self.qc, backend=self.backend, shots=shots).result().get_counts()
 
         try:
-            xpiled_qc = transpile(self.qc, backend=backend, optimization_level=opt_level)
+            xpiled_qc = transpile(
```

Comment:
Since we're transpiling here I think it's probably worth changing L427 to be `backend.run(xpiled_qc, shots=shots)` instead of `execute()`. Mostly because calling `execute()` will retranspile the circuit with level 1 and default options. This should mostly be a no-op but it is probably better to explicitly just run the circuit on the backend directly and avoid an additional transpilation.

Reply:
Ah that's a great idea. Done in 0ddd5478a6a541f08e5ba1ce21ee74d3563d09ae.